### PR TITLE
PS-5405: binlog_skip_flush_commands

### PR DESCRIPTION
**Pull request by borys-belinsky**

### CODE REVIEW

File: doc/source/flexibility/binlogging_replication_improvements.rst [hunk 1]
```diff
@@ -108,10 +109,57 @@ condition. This fix has been ported from the upstream bug report
...
 
   * :rn:`5.6.13-60.5`
     Fix ported from |Percona Server| 5.5
 
+.. _percona-server.binary-log.flush.writing:
 
+Writing ``FLUSH`` Commands to the Binary Log 
+================================================================================
 
+``FLUSH`` commands, such as such as ``FLUSH SLOW LOGS``, are not written to the
```
Comment (percona-ysorokin) [inline doc/source/flexibility/binlogging_replication_improvements.rst]: I would not combine the following 2 paragraphs

```
``FLUSH`` commands, such as such as ``FLUSH SLOW LOGS``, are not written to the
binary log if the system variable :variable:`binlog_skip_flush_commands` is set
to **ON**.

In addition, the following changes in the readonly / super_readonly behavior were implemented:
- When ``read_only`` is set to ON, any FLUSH XXX command executed by a normal user (w/o SUPER privilege) will not be written to the binary log regardless of the value of the |bsfc| variable.
- When ``super_read_only`` is set to ON, any FLUSH XXX command executed by any user (even by those with SUPER privilege) will not be written to the binary log regardless of the value of the |bsfc| variable.

An attempt to run a ``FLUSH`` command without either ``SUPER`` or ``RELOAD``
privileges, results in the ``ER_SPECIFIC_ACCESS_DENIED_ERROR`` exception
regardless of the value of the |bsfc| variable.
```

File: doc/source/flexibility/binlogging_replication_improvements.rst [hunk 2]
```diff
@@ -12,25 +12,26 @@ Temporary tables and mixed logging format
...
 
   * :rn:`5.6.21-70.1`
     Fix implemented in |Percona Server| 5.6
 
 Details:
-********
+--------------------------------------------------------------------------------
 
-Mixed binary logging format supported by |Percona Server| means that
+The Mixed binary logging format supported by |Percona Server| means that
```
Comment (laurynas-biveinis) [inline doc/source/flexibility/binlogging_replication_improvements.rst]:
```suggestion
The mixed binary logging format supported by |Percona Server| means that
```

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: doc/source/flexibility/binlogging_replication_improvements.rst (hunk 1 above)
Comment (borys-belinsky) [reply doc/source/flexibility/binlogging_replication_improvements.rst]: Ready for review

File: doc/source/flexibility/binlogging_replication_improvements.rst [hunk 3]
```diff
@@ -108,10 +109,62 @@ condition. This fix has been ported from the upstream bug report
...
+
+When set to **ON**, ``FLUSH ...`` commands are not written to the binary
+log. See :ref:`percona-server.binary-log.flush.writing` for more information
+about what else affects the writing of ``FLUSH`` commands to the binary log.
+
+.. note::
+
+   ``FLUSH LOGS``, ``FLUSH BINARY LOGS``, ``FLUSH TABLES WITH READ LOCK``, and
+   ``FLUSH TABLES ... FOR EXPORT`` are not written to the binary log no matter
+   what the value of the :variable:`binlog_skip_flush_commands` variable
```
Comment (percona-ysorokin) [inline doc/source/flexibility/binlogging_replication_improvements.rst]:
```suggestion
   what value :variable:`binlog_skip_flush_commands` variable
```

File: doc/source/flexibility/binlogging_replication_improvements.rst [hunk 4]
```diff
@@ -108,10 +109,62 @@ condition. This fix has been ported from the upstream bug report
...
+   contains. The ``FLUSH`` command is not recorded to the binary log and value
+   of :variable:`binlog_skip_flush_commands` is also ignored if the ``FLUSH``
+   command was run with the ``NO_WRITE_TO_BINLOG`` keyword (or its alias
```
Comment (percona-ysorokin) [inline doc/source/flexibility/binlogging_replication_improvements.rst]:
```suggestion
   command is run with the ``NO_WRITE_TO_BINLOG`` keyword (or its alias
```

File: doc/source/flexibility/binlogging_replication_improvements.rst (hunk 3 above)
Comment (borys-belinsky) [reply doc/source/flexibility/binlogging_replication_improvements.rst]: Ready for review

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Pull request merged**